docs/specs/bind-unbind-workflow.md

```diff
@@ -0,0 +1,32 @@
+## Bind / Unbind New Work flow
+
+The goal of this new work flow is to give more control over the actions that the database takes on bind and unbind to the APB author. This means that we want the apb to be launched and to run the bind or unbind actions.
+
+While this is still using existing workflow pieces (i.e extracting credentials, saving those credentials), it is also creating a new workflow. This is an enhancement on the current design and will allow for the APB to choose which workflow it would prefer to follow. 
+
+### Default Settings Needed
+* set `launch_apb_on_bind: true` - If this is set to false we will behave the exact same as today, handing back the credentials received at provision to the service catalog.
+
+
+### Brokers Responsibility
+* The broker will save the credentials for each bind. this will be saved in etcd under the `extracted_credentials` directory. The key will be the binding id.
+* At provision time, the broker should inform the APB of the launch_apb_on_bind value. The apb can then choose what it wants to do. 
+* The broker will be responsible for providing the provisioned credentials (extracted credentials given at provision time) to the APB as extra vars during bind and unbind. Example
+  - `--extra-vars '{..., "_apb_db_params:{}`
+* The broker will be responsible for deleting the bindings extracted credentials in etcd during unbind.
+* The broker will be responsible for deleting the bindings extracted credentials in etcd during deprovision.
```

Comment:
Our concerns on this were raised in IRC. The broker should never receive requests to delete potentially referenced resources. I.E. Delete ServiceInstance with outstanding bindings. That sounds like a hard error to me. At least until the spec suggests otherwise.

If this occurs, 4xx. Bad request.